src/dftd4/param.f90

```diff
@@ -67,7 +67,8 @@ subroutine get_rational_damping(functional, param, s9)
    integer :: is, id
    logical :: mbd
 
-   mbd = merge(s9 /= 0.0_wp, .true., present(s9))
+   mbd = .false.
+   if (present(s9) .and. s9 /= 0.0_wp) mbd = .true.
```

Comment:
The condition may not be short-circuited.

```suggestion
   if (present(s9)) mbd = s9 /= 0.0_wp
```

Reply:
You are right. Thanks for the hint.